minio/minio
# flatten out audit tags, do not send as free-form

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
flatten out audit tags, do not send as free-form

## Motivation and Context
move away from map[string]interface{} to map[string]string to 
simplify the audit, and also provide concise information.

avoids large allocations under load() and reduces the amount 
of audit information generated, as the current implementation 
was a bit free-form. instead, all data structures must be flattened.

## How to test this PR?
Enable ILM with older mtime objects and then observe more extensive
output in `mc admin trace --call ilm --json`. This time, we provide
the rule-id etc the ILM is being invoked on.

## Types of changes

## Checklist:
